tests/lib/rules/no-empty-function.js

```diff
@@ -267,6 +269,16 @@ ruleTester.run("no-empty-function", rule, [
         {
             code: "var foo = () => 0;",
             parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: "const foo = { async method() {} }",
+            options: [{ allow: ["asyncMethods"] }],
+            parserOptions: { ecmaVersion: 8 }
+        },
+        {
+            code: "async function a(){}",
+            options: [{ allow: ["asyncFunctions"] }],
+            parserOptions: { ecmaVersion: 8 }
```

Comment:
Could you move these test cases in the reduced `array` for consistency?
I think test cases in here are for the `non-empty` function cases. e.g.`var foo = () => 0;`

https://github.com/eslint/eslint/blob/f1525dc45dfdbbe31e724671270785b41cffc6bd/tests/lib/rules/no-empty-function.js#L90-L95

Reply:
what I understood from this test setup is that these cases are invalid cases and these will be converted into valid and invalid cases. right !

Reply:
> Would you mind updating the documentation also? - [no-empty-function](https://github.com/eslint/eslint/blob/master/docs/rules/no-empty-function.md#options)

sure !